pkg/languagedetection/detector.go

```diff
@@ -0,0 +1,46 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package languagedetection
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+)
+
+type Language struct {
+	Name string
+}
+
+// Maps the prefix to the language name
```

Comment:
I think it's the other way around, you're mapping a language name to a prefix no ?

Reply:
Not as far as I understand it. It's a map indexed by prefix to language name. For example in the future we may add these prefixes:
```
irb    => ruby
ruby => ruby
```